src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx

```diff
@@ -0,0 +1,67 @@
+import { Checkbox, FormControlLabel } from '@mui/material';
+import Box from '@mui/material/Box';
+import React, { useEffect, useState } from 'react';
+
+import { GCSConfig, IntegrationConfig } from '../../../utils/integrations';
+import { IntegrationTextInputField } from './IntegrationTextInputField';
+
+const Placeholders: GCSConfig = {
+  bucket: 'aqueduct',
+  credentials_path: '',
+  use_as_storage: '',
+};
+
+type Props = {
+  setDialogConfig: (config: IntegrationConfig) => void;
+};
+
+export const GCSDialog: React.FC<Props> = ({ setDialogConfig }) => {
+  const [bucket, setBucket] = useState<string>(null);
```

Comment:
I just merged #320 and you might want to rebase to follow the new pattern. You could refer to any other dialog like snowflake as example. Happy to meet offline to help!

Reply:
Rebased! The new method is cool :)